packages/integration-test-app/windows/integrationtest/TestHostHarness.cpp

```diff
@@ -85,70 +81,81 @@ void TestHostHarness::OnInstanceLoaded(const InstanceLoadedEventArgs &args) noex
       });
 }
 
-winrt::fire_and_forget TestHostHarness::StartListening() noexcept {
+IAsyncAction TestHostHarness::StartListening() noexcept {
   VerifyElseCrash(m_dispatcher.HasThreadAccess());
 
-  auto listenResult = co_await m_commandListener->StartListening();
+  auto cancellationToken = co_await winrt::get_cancellation_token();
 
-  switch (listenResult) {
-    case ListenResult::Success:
-      co_return;
+  // Keep on polling for clients until we die
+  while (!cancellationToken()) {
+    auto processReqsTask = m_rpcServer.ProcessAllClientRequests(8305 /*port*/, 50ms /*pollInterval*/);
 
-    case ListenResult::AddressInUse:
-      ShowJSError("Test harness address is already in use. Is there another instance running?");
-      co_return;
+    // Foward cancellation to also cancel processing reqs (setting callback replaces the last)
+    cancellationToken.callback([processReqsTask]() noexcept { processReqsTask.Cancel(); });
+    co_await processReqsTask;
+  }
+}
 
-    case ListenResult::OtherError:
-      ShowJSError("The test harness ran into an unexpected error trying to listen for commands.");
-      co_return;
+winrt::NodeRpc::Handler TestHostHarness::CreateRpcHander() noexcept {
+  winrt::NodeRpc::Handler handler;
 
-    default:
-      ShowJSError("Unexpected ListenResult from TestCommandListener.");
-      co_return;
-  }
+  handler.BindAsyncOperation(
```

Comment:
>handler [](start = 2, length = 7)

Just a generic comment... We need some helper pattern here.. I understant that cpp should be explicit, but the logic to handle weakThis vs strongThis is getting quire redicuouls in my view...

Reply:
Yes... Vlad has a `MakeWeakMemberFunctor` method in ReactHost code that I think is a good pattern for Mso smart pointers.   Not sure anyone has done the same for cppwinrt.

Separate question of sharing between projects. I have a source copy of a utility from react-native-windows package in NodeRpc package bc we needed the same 50 lines. We have plug-in native projects from JS with auto-linking, but It would be overkill for a header only library.

Maybe just vcxshared item lists in a package folder without package.json?

Reply:
That sounds like a reasonable plan.